libcxx/include/variant

```diff
@@ -834,21 +837,24 @@ class _LIBCPP_TEMPLATE_VIS __move_constructor;
       using __base_type::__base_type;                                                                                  \
       using __base_type::operator=;                                                                                    \
                                                                                                                        \
-      __move_constructor(const __move_constructor&)            = default;                                              \
-      move_constructor ~__move_constructor()                   = default;                                              \
```

Comment:
Maybe adding `move_constructor;` would help clang-format put those on different lines? That would really increase readability. Otherwise perhaps even `/*                                   */` would help.

Reply:
I tried and it indeed made clang-format happy. However, gcc complained that
```
error: extra ';' after member function definition [-Werror,-Wextra-semi]
```

I tried various things and none of them can format it nicely (partially due to the fact that we are inside a macro definition).

Another thing we can do is to add 
```
// clang-format off
// define macro here
// clang-format on
```